pkg/log/checkpoint/connectivity.go

```diff
@@ -0,0 +1,33 @@
+package checkpoint
+
+import (
+	"net"
+)
+
+type dialer interface {
+	Dial(network, address string) (net.Conn, error)
+}
+
+func testConnections(dialer dialer, hosts ...string) map[string]interface{} {
+	results := make(map[string]interface{})
+
+	for _, host := range hosts {
+		if err := testConnection(dialer, host); err != nil {
+			results[host] = err.Error()
+		} else {
+			results[host] = "successful tcp connection over 443"
+		}
+	}
+
+	return results
+}
+
+func testConnection(dialer dialer, host string) error {
+	hostPort := net.JoinHostPort(host, "443")
```

Comment:
This is not guaranteed, but maybe fine to roll with.

Reply:
yea, I wrote this part before we started parsing the options, I can see now in the opts we are provided a port, I can update this to accept a port as well